### PR TITLE
NODE-1522: tests for matcherFeeAssetId

### PR DESCRIPTION
**Pull request by finnoio**

### CODE REVIEW

File: it/src/test/scala/com/wavesplatform/it/sync/smartcontract/ExchangeWithContractsSuite.scala [hunk 1]
```diff
@@ -200,6 +204,125 @@ class ExchangeWithContractsSuite extends BaseTransactionSuite with CancelAfterFa
     )
   }
 
+  test("positive - exchange tx with asset/smartasset matcherFee") {
```
Comment (monroid) [inline it/src/test/scala/com/wavesplatform/it/sync/smartcontract/ExchangeWithContractsSuite.scala]: Was discussed offline

File: it/src/test/scala/com/wavesplatform/it/sync/matcher/MatcherFeeAssetSuite.scala [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+            val aliceWavesPair = AssetPair(assetInPair, Waves)
+            val ts = ntpTime.correctedTime()
+            val expirationTimestamp = ts + Order.MaxLiveTime - 10000
+            val amount = 100
+            var buy = None: Option[Order]
+            var sell = None: Option[Order]
+
+            percentAssetType match {
+              case "amount" =>
+                buy = Some(Order.buy(aliceAcc, matcherNode.privateKey.publicKey, aliceWavesPair, amount, Order.PriceConstant, ts,
```
Comment (DrBlast) [inline it/src/test/scala/com/wavesplatform/it/sync/matcher/MatcherFeeAssetSuite.scala]: "Some" is not used

File: it/src/test/scala/com/wavesplatform/it/sync/matcher/MatcherFeeAssetSuite.scala [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+            val aliceOrderIdFill = matcherNode
+              .placeOrder(buy.get)
+              .message
+              .id
+            val bobOrderIdFill = matcherNode
+              .placeOrder(sell.get)
+              .message
+              .id
+
+            orderStatus(aliceAcc, aliceWavesPair, aliceOrderIdFill, "Filled")
```
Comment (DrBlast) [inline it/src/test/scala/com/wavesplatform/it/sync/matcher/MatcherFeeAssetSuite.scala]: add balances check

**Comment by phearnot** [comment]
Something's broken. We'll just create a new branch & PR with relevant changes.

**Pull request closed without merging**